packages/eslint-plugin/src/util/misc.ts

```diff
@@ -241,20 +241,66 @@ type NodeWithKey =
   | TSESTree.TSAbstractPropertyDefinition;
 
 /**
- * Gets the key being accessed or declared, such as `value` in
- * `x.value`, `x['value']`,
- * or even `const v = 'value'; x[v]` (or optional variants thereof).
+ * Gets a member being accessed or declared if its value can be determined statically, and
+ * resolves it to the string or symbol value that will be used as the actual member
+ * access key at runtime. Otherwise, returns `undefined`.
+ *
+ * ```ts
+ * x.member // returns 'member'
+ * ^^^^^^^^
+ *
+ * x?.member // returns 'member' (optional chaining is treated the same)
+ * ^^^^^^^^^
+ *
+ * x['value'] // returns 'value'
+ * ^^^^^^^^^^
+ *
+ * x[Math.random()] // returns undefined (not a static value)
+ * ^^^^^^^^^^^^^^^^
+ *
+ * arr[0] // returns '0' (NOT 0)
+ * ^^^^^^
+ *
+ * arr[0n] // returns '0' (NOT 0n)
+ * ^^^^^^^
+ *
+ * const s = Symbol.for('symbolName')
+ * x[s] // returns `Symbol.for('symbolName')` (since it's a static/global symbol)
+ * ^^^^
+ *
+ * const us = Symbol('symbolName')
+ * x[us] // returns undefined (since it's a unique symbol, so not statically analyzable)
+ * ^^^^^
+ *
+ * var object = {
+ *     1234: '4567', // returns '1234' (NOT 1234)
+ *     ^^^^^^^^^^^^
+ *     method() { } // returns 'method'
+ *     ^^^^^^^^^^^^
+ * }
+ *
+ * class WithMembers {
+ *     foo: string // returns 'foo'
+ *     ^^^^^^^^^^^
+ * }
+ * ```
  */
 function getStaticMemberAccessValue(
   node: NodeWithKey,
   { sourceCode }: RuleContext<string, unknown[]>,
 ): string | symbol | undefined {
   const key =
     node.type === AST_NODE_TYPES.MemberExpression ? node.property : node.key;
-  if (!node.computed) {
-    return key.type === AST_NODE_TYPES.Literal
-      ? String(key.value)
-      : (key as TSESTree.Identifier | TSESTree.PrivateIdentifier).name;
+  const { type } = key;
+  if (type === AST_NODE_TYPES.Literal) {
```

Comment:
(question, no specific action/resolution implied) is this conditional necessary? Or will it just be caught by the `getStaticValue`?

Reply:
checked locally and all the tests pass if I comment it out 🤷

Reply:
good catch, removed!